src/hex_tarball.erl

```diff
@@ -439,6 +521,74 @@ gzip_no_header(Uncompressed) ->
         zlib:close(Zstream)
     end.
 
+%%====================================================================
+%% Meta-Validation
+%%====================================================================
+-spec is_valid_meta(metadata()) -> {ok, list()} | {error, map()}.
```

Comment:
Warnings and errors shouldn't be exclusive. Can the return value instead be `{list(), list()}`? With `{[], []}` meaning no errors and no warnings.

Reply:
That makes sense to me.